### ISSUE TITLE
EFFECT_MORNING_SUN and EFFECT_MOONLIGHT are redundant

### ISSUE THREAD

**Issue by BuffelSaft** [rh-hideout/pokeemerald-expansion#2444]
This seems to be junk leftover from gen 2, where these moves were actually different (they each healed different amounts based on the time of day). From gen 3 onwards, they're identical to EFFECT_SYNTHESIS, so they could all use the same move effect.

This would simplify checks like this:
```           
if (HasMoveEffect(battlerDef, EFFECT_MORNING_SUN) 
|| HasMoveEffect(battlerDef, EFFECT_SYNTHESIS) 
|| HasMoveEffect(battlerDef, EFFECT_MOONLIGHT))
```
into `if (HasMoveEffect(battlerDef, EFFECT_SYNTHESIS)`

**Comment by AsparagusEduardo** [comment]
Comparing the Bulbapedia page, Moonlight doesn't state what happens in Strong Winds, compared to the other two moves.

**Comment by AsparagusEduardo** [comment]
Confirmed that in USUM all three moves have the exact same effect.

**Comment by DizzyEggg** [comment]
We could always keep the effects and add a config for the GEN2 behaviour.

**Comment by AsparagusEduardo** [comment]
> We could always keep the effects and add a config for the GEN2 behaviour.

I remember there were discussions about emulating Gen 1/2's battle quirks via config once we caught up to the latest Gen and had nothing else to do, but didn't go too far.

At one point I tested the possibility of adding mid-gen configs, including Gen 1/2 ones, but ended up being too convoluted. Maybe we could start looking into it again 👀

**Comment by DizzyEggg** [comment]
> > We could always keep the effects and add a config for the GEN2 behaviour.
> 
> I remember there were discussions about emulating Gen 1/2's battle quirks via config once we caught up to the latest Gen and had nothing else to do, but didn't go too far.
> 
> At one point I tested the possibility of adding mid-gen configs, including Gen 1/2 ones, but ended up being too convoluted. Maybe we could start looking into it again 👀

Yeah, I noticed you even started with one PR for Shuckle gen2 behavior. I think having them as available, but set to FALSE by default, is a good option.

**Comment by Pawkkie** [comment]
Given that we're now in the business of retroactively adding gen 1 and gen 2 configs as contributors feel the interest (it's been a while since this issue lol), I think we should close this issue and instead make a feature request for gen 2 Morning Sun / Moonlight configs.

Would like at least one other person to agree before I go ahead and do that :D

**Comment by i0brendan0** [comment]
> Given that we're now in the business of retroactively adding gen 1 and gen 2 configs as contributors feel the interest (it's been a while since this issue lol), I think we should close this issue and instead make a feature request for gen 2 Morning Sun / Moonlight configs.
> 
> Would like at least one other person to agree before I go ahead and do that :D

Does it have to be someone working on the project to agree? Because with the fact that there's plenty of Gen 1/2 specific configuration options already and that I've heard a Day/Night system is in the works, this would be something that could be useful to someone wanting that system or if they just changed the Day/Night times in the code. At least that's my two cents on the matter.

**Comment by Pawkkie** [comment]
Closed, reversing this to instead make a feature request for gen 2 behaviour :)
